### ISSUE TITLE
Cannot install gosu with ruby-1.9.3-p551 and rbenv (homebrew)

### ISSUE THREAD

**Issue by Infotaku** [gosu/gosu#321]
Hi,

I'm having an issue installing gosu for ruby 1.9.3 (which seems to be required by releasy for windows builds).

> The Gosu gem requires some libraries to be installed system-wide.
>See the following site for a list:
>https://github.com/jlnr/gosu/wiki/Getting-Started-on-OS-X
>
>creating Makefile
>
>make
>compiling stb_vorbis.c
>compiling gosu_wrap.cxx
>gosu_wrap.cxx:1852:10: fatal error: 'string' file not found
>#include <string>
>         ^
>1 error generated.
>make: *** [gosu_wrap.o] Error 1

I've installed everything from the guide but still can't install gosu.

Any ideas ?

Thank you !

**Comment by jlnr** [comment]
That's unexpected, but I can reproduce the issue and will look into it. In the meantime, you should probably experiment with Releasy on Windows to see that it actually works :) - otherwise you're limiting yourself to an old version of Ruby for no practical reason.

**Comment by jlnr** [comment]
Oh my. For some reason, Ruby 1.9.3 seems to have inherited this bug?

https://bugs.ruby-lang.org/issues/8315

In any case, the CXXFLAGS are not set in the generated Makefile:

`CXXFLAGS = $(CFLAGS) $(cxxflags)`

should be (from Ruby 2.0.0 in rbenv):

`CXXFLAGS = $(CCDLFLAGS) $(cxxflags) -x objective-c++ -U_GLIBCXX_DEBUG -U_GLIBCXX_DEBUG_PEDANTIC -std=gnu++11 -stdlib=libc++ -I/usr/local/include/SDL2 -D_THREAD_SAFE $(ARCH_FLAG)`

Using C++ and Objective C in RubyGems is such a hot mess...

**Comment by jlnr** [comment]
I've pushed a fix which will be included in v0.10.7.

You can already give it a try by checking out Gosu from source, then running:
```sh
gem install rake-compiler
export GOSU_RELEASE_VERSION=0.10.6
rake gem
gem install pkg/gosu-0.10.6.gem```

Hope that helps.

**Comment by Infotaku** [comment]
Thank you for the fix !
